manager/usermanager/user.go

```diff
@@ -21,8 +21,9 @@ func IsUserExists(userStore *store.UserStore, user *models.UserCredentials) (boo
 		return false, err
 	}
 
-	if !exists && user.Email != "" {
-		_, err := userStore.GetUser(bson.M{"email": user.Email})
+	if !exists && user.UnverifiedEmail != "" {
+		// Check if the supplied email is already registered with some other user
+		_, err := userStore.GetUser(bson.M{"email": user.UnverifiedEmail})
```

Comment:
Why are we checking with unverifiedEmail? When will this code be called?

Reply:
This code will be called at the time of user creation. Since in the request we will have the email assigned to `unverified_email` field. So we need to check whether this eail is existing or not. E.g. 
The request is like below to `/v1/signup`
```
{
"unverfied_email": "example@gmail.com"
"password": "example@123",
"name": "Example"
}
```
So now `example@gmail.com` will be checked against the `email` field of existing users.

Reply:
That's a part of verifyEmail(), right?
Or is this the case for admin creating users?

Reply:
Before `sendVerificationLink()` a user will be created in the database. So before creating this check is there for self signing up only.

Reply:
Okay